base/src/main/java/com/fasterxml/jackson/jaxrs/base/ProviderBase.java

```diff
@@ -589,14 +589,21 @@ public void writeTo(Object value, Class<?> type, Type genericType, Annotation[]
                     //    since forcing of type will then force use of content serializer, which is
                     //    generally not the intent. Fix may require addition of functionality in databind
 
-                    rootType = writer.getTypeFactory().constructType(genericType);
+					TypeFactory typeFactory = writer.getTypeFactory();
+					rootType = typeFactory.constructType(genericType);
+					Class<?> rawClass = rootType.getRawClass();
                     /* 26-Feb-2011, tatu: To help with [JACKSON-518], we better recognize cases where
                      *    type degenerates back into "Object.class" (as is the case with plain TypeVariable,
                      *    for example), and not use that.
                      */
-                    if (rootType.getRawClass() == Object.class) {
-                        rootType = null;
-                    }
+					if (rawClass == Object.class) {
+						rootType = null;
+					} else if (type != rawClass && !rootType.isCollectionLikeType() && !rootType.isMapLikeType()
+							&& rootType.hasGenericTypes()) {
+						List<JavaType> typeParameters = rootType.getBindings().getTypeParameters();
```

Comment:
Instead of this a better way would be:

    rootType = typeFactory.constructSpecializedType(rootType, rawClass);

which should do the right thing, as long as `rawClass` is a proper sub-class of type-erased class of `rooType`.
It often ends up doing what code here does, but not necessarily (there are all kinds of possible edge cases).

Would it be possible to see if code works with that change? (just because it should work fine doens't always mean it will :) )